src/dali/abstract_pair_converter_plugin.py

```diff
@@ -53,6 +67,31 @@ class AssetPairAndTimestamp(NamedTuple):
     exchange: str
 
 
+class MappedGraph(Graph[ValueType]):
+    def __init__(self, vertexes: Optional[List["Vertex[ValueType]"]] = None) -> None:
+        super().__init__(vertexes)
+        self.__name_to_vertex: Dict[str, Vertex[ValueType]] = {vertex.name: vertex for vertex in vertexes} if vertexes else {}
+
+    def get_vertex(self, name: str) -> Optional[Vertex[ValueType]]:
+        return self.__name_to_vertex.get(name)
+
+    def get_or_set_vertex(self, name: str) -> Vertex[ValueType]:
+        old_vertex: Optional[Vertex[ValueType]] = self.get_vertex(name)
+        if old_vertex:
+            return old_vertex
+
+        new_vertex: Vertex[ValueType] = Vertex[ValueType](name=name)
+        super().add_vertex(new_vertex)
```

Comment:
I think we need to override `add_vertex()` too: it can call its super and then add the vertex to the dict. Then here we can just call `add_vertex()`.